### PR TITLE
Add option to disable masking from RectTransformColliderFitter

### PR DESCRIPTION
**Pull request by camnewnham**
## Overview
This PR adds an option to disable the masking behaviour of `RectTransformColliderFitter`. This is useful when a component needs to extend beyond a masked region (i.e. pop out of a `ScrollRect`.)

## Changes
- Fixes this issue where components that pop out of a `ScrollRect` have their colliders disabled. 
dropdown

:)

### CODE REVIEW

File: com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs [hunk 1]
```diff
@@ -52,6 +52,36 @@ public BoxCollider ThisCollider
         [Tooltip("2D padding around the RectTransform.")]
         private Vector2 padding = Vector2.zero;
 
+        [SerializeField]
+        [Tooltip("Whether the collider should be affected by Rect2DMasks in parent components")]
+        private bool masked = true;
```
Comment (Zee2) [inline com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs:57]: Style suggestion; we generally try to name the private field and public property the same for clarity. I'd recommend naming these `applyMask` and `ApplyMask`, or `UseMask`, or something along those lines! Thanks for this, it's a great feature to have.

File: com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs [hunk 2]
```diff
@@ -203,6 +238,10 @@ private void Fit()
                         }
                     }
                 }
+                else if (!ApplyMask && thisCollider != null && !thisCollider.enabled)
+                {
+                    thisCollider.enabled = true;
```
Comment (Zee2) [inline com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs:243]: Why is this necessary? Colliders shouldn't be disabled in the first place.

**Review by Zee2 — CHANGES_REQUESTED** [review]

**Comment by Zee2** [comment]
Your popup menu looks fantastic, by the way! Feel free to open a PR for that too if you'd like!

I'm working on my own dropdown in another branch, but I'd love to take a look at yours as well.

File: com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs (hunk 1 above)
Comment (camnewnham) [reply com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs]: Thanks, updated.

File: com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs (hunk 2 above)
Comment (camnewnham) [reply com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs:243]: This is necessary to correctly support changes during runtime, in this scenario (albeit a rare case presumably):
1. Mask is enabled. Collider is disabled because it doesn't intersect the mask.
2. `UseMask` is toggled (either by editor GUI or script)
3. The collider needs to be re-enabled during the next `Fit()` as the mask should no longer apply.

**Comment by camnewnham** [comment]
> I'm working on my own dropdown in another branch, but I'd love to take a look at yours as well.

I'd be happy to - though there are a couple of things to resolve before it'd be flexible enough and suitable for general consumption:
1. https://github.com/microsoft/MixedRealityToolkit-Unity/pull/10795
2. It's very easy to accidentally press _through_ a popup and trigger the button or another piece of UI behind - can you suggest a strategy to work around this?
3. It needs the equivalent of a "clickaway" listener  - any suggestions?

File: com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs (hunk 2 above)
Comment (Zee2) [reply com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs]: Ah, gotcha.

**Comment by Zee2** [comment]
@camnewnham 

> It's very easy to accidentally press through a popup and trigger the button or another piece of UI behind - can you suggest a strategy to work around this?
> It needs the equivalent of a "clickaway" listener - any suggestions?

XRI doesn't really do "modality"; i.e., there's no way to push or pop layers of UI modally, so neither of these are really that easy to do with XRI. For the first one, I would suggest literally bringing the popup towards the user and increase the Z separation between the foreground popup and background.

Clickaway events are similarly tricky... clicking "away" implies that the thing itself has some sort of modal focus. In the future, we may ask XRI folks to either implement this sort of modality, or perhaps implement our own support for it as a subclass of the XRInteractionManager (or at least add "clickaway" event handling to the XRInteractionManager).

https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@2.0/api/UnityEngine.XR.Interaction.Toolkit.XRInteractionManager.html

File: com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs [hunk 3]
```diff
@@ -179,7 +213,12 @@ private void Fit()
 
                 Rect computedRect = attachedRectTransform.rect;
 
-                if (mask != null)
+                if (useMask != UseMask)
```
Comment (RogPodge) [inline com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs:216]: Won't this case never happen because UseMask's getter returns useMask?

**Review by RogPodge — CHANGES_REQUESTED** [review]

File: com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs (hunk 3 above)
Comment (RogPodge) [reply com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs:216]: Besides this change, the branch looks good to me and should be ready to merge.

**Comment by camnewnham** [comment]
I've updated this to add some clarity. The previous approach using a setter was slightly confusing because runtime changes in editor (toggling `useMask`) wouldn't be triggered, which made the logic harder to read.

Instead:
- Use the serialized editor-friendly property `useMask` to determine the intent (should mask be applied or not)
- Cache value `maskApplied` to know the current state of the mask (is the mask currently applied)
- During a `Fit()` operation, ensure that the mask is applied or removed as required

File: com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs [hunk 4]
```diff
@@ -74,6 +78,8 @@ public Vector2 Padding
 
         private ScrollRect scrollRect;
 
+        private bool maskApplied = false;
```
Comment (Zee2) [inline com.microsoft.mrtk.uxcore/Button/RectTransformColliderFitter.cs:81]: Could you please move this to be a private return var inside your UpdateMask function?
